src/main/scala/cromwell/engine/backend/jes/JesEngineFunctions.scala

```diff
@@ -0,0 +1,81 @@
+package cromwell.engine.backend.jes
+
+import java.nio.file.Path
+import java.security.MessageDigest
+
+import cromwell.binding.values._
+import cromwell.engine.EngineFunctions
+import cromwell.util.GcsUtil
+import cromwell.util.google.GoogleCloudStoragePath
+
+import org.apache.commons.codec.binary.Base64
+
+import scala.util.{Success, Try}
+
+/**
+ * Implementation of engine functions for the JES backend.
+ */
+case class JesEngineFunctions(callDir: GoogleCloudStoragePath, jesConnection: JesInterface) extends EngineFunctions {
+
+  private def readFromPath(value: String): String = {
+    val tryParse = GoogleCloudStoragePath.parse(value)
+    val gcsPathToUse: GoogleCloudStoragePath = if (tryParse.isSuccess) { tryParse.get } else { GoogleCloudStoragePath(gcsPathFromAnyString(value)) }
+    jesConnection.storage.slurpFile(gcsPathToUse)
+  }
+  
+  /**
+   * Read the entire contents of a file from the specified `WdlValue`, where the file can be
+   * specified either as a path via a `WdlString` (with magical handling of "stdout"), or
+   * directly as a `WdlFile`.
+   *
+   * @throws UnsupportedOperationException for an unrecognized file reference, as this is intended
+   *                                       to be wrapped in a `Try`.
+   */
+  private def fileContentsToString(value: WdlValue): String = {
+    value match {
+      case f: WdlFile => {
```

Comment:
You don't need braces in case statements. Also when they're real short like this I think it's better to just put them on one line